routing/probability_bimodal.go

```diff
@@ -0,0 +1,536 @@
+package routing
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/go-errors/errors"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+const (
+	// DefaultBimodalScaleMsat is the default value for BimodalScaleMsat in
+	// BimodalConfig. It describes the distribution of funds in the LN based
+	// on empirical findings. We assume an unbalanced network by default.
+	DefaultBimodalScaleMsat = lnwire.MilliSatoshi(300_000_000)
+
+	// DefaultBimodalNodeWeight is the default value for the
+	// BimodalNodeWeight in BimodalConfig. It is chosen such that past
+	// forwardings on other channels of a router are only slightly taken
+	// into account.
+	DefaultBimodalNodeWeight = 0.2
+
+	// DefaultBimodalDecayTime is the default value for BimodalDecayTime.
+	// We will forget about previous learnings about channel liquidity on
+	// the timescale of about a week.
+	DefaultBimodalDecayTime = time.Duration(7*24) * time.Hour
```

Comment:
This is indeed spectacular

Reply:
`string("are you talking about the extra type cast")`?

Reply:
Hehe, no increasing decay to a week.

Reply:
I think it makes sense to have a longer decay time. In the apriori model, we don't forget about successes, only about failures, but here failures and successes are treated symmetrically, so it would be a pity to forget about those in short time (we could make those asymmetric if we wanted to as discussed earlier).

Reply:
Yeah I think worth it to give this a shot, it's also configurable so people can set it to other values. In retrospect maybe using a 12 hour decay (user offline for a week and has no memory) was _too_ conservative.

Reply:
Related: https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-February/003842.html